### PR TITLE
Aaron/origin alignment 

### PR DESCRIPTION
**Pull request by aaroncherian**
Code for aligning stuff with the origin

The debug option for the origin alignment (those four skeleton plots I showed in lab meeting) is set to false by default to make sure it doesn't interrupt the run, just an fyi

### CODE REVIEW

File: freemocap/fmc_origin_alignment.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from pathlib import Path 
+from rich.progress import track 
+
+def align_skeleton_with_origin(session, mediapipe_skeleton_data_XYZ, good_clean_frame_number):
```
Comment (jonmatthis) [inline freemocap/fmc_origin_alignment.py:8]: This'll probably want to be a `class` at some point

File: freemocap/fmc_origin_alignment.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+    #skeleton_data_path = session.dataArrayPath/'mediaPipeSkel_3d_smoothed.npy'
+    debug = True
+    right_heel_index = 30
+    right_toe_index = 32
+    left_heel_index = 29
+    left_toe_index = 31
```
Comment (jonmatthis) [inline freemocap/fmc_origin_alignment.py:15]: and we'll pull these numbers from some `MediapipeDefinitions` dataclass

File: freemocap/fmc_origin_alignment.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        """Put two vectors in, make a normal vector"""
+        normal_vector = np.cross(vector1,vector2)
+        return normal_vector
+
+    def create_unit_vector(vector): 
+        """Take in a vector, make it a unit vector"""
+        unit_vector = vector/np.linalg.norm(vector)
+        return unit_vector
+
+    def calculate_skewed_symmetric_cross_product(cross_product_vector):
```
Comment (jonmatthis) [inline freemocap/fmc_origin_alignment.py:39]: is this the part you got from stack overflow?

Comment (jonmatthis) [reply freemocap/fmc_origin_alignment.py:39]: in general, code like this that encapsulates a lot of logic in a single call is hard to debug. Better to caluculate each row independently and then stack them to get the final output

File: freemocap/fmc_origin_alignment.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+        skew_symmetric_cross_product = np.array([[0, -cross_product_vector[2], cross_product_vector[1]],
+                                                [cross_product_vector[2], 0, -cross_product_vector[0]],
+                                                [-cross_product_vector[1], cross_product_vector[0], 0]])
+        return skew_symmetric_cross_product
+
+    def calculate_rotation_matrix(vector1,vector2):
+        """Put in two vectors to calculate the rotation matrix between those two vectors"""
+        #based on the code found here: https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d"""
```
Comment (jonmatthis) [inline freemocap/fmc_origin_alignment.py:47]: ah, gotcha 👍

**Review by jonmatthis — APPROVED** [review]
I don't have time to dig in to all the nitty gritty of the math, but in general the code looks nice and clean! Super cool to see you developing some fluency working with the data on this level :D

File: freemocap/fmc_origin_alignment.py (hunk 3 above)
Comment (aaroncherian) [reply freemocap/fmc_origin_alignment.py:39]: image

what I got from stack exchange was this equation for rotating some unit vector a onto unit vector b

Comment (aaroncherian) [reply freemocap/fmc_origin_alignment.py:39]: https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d

Comment (jonmatthis) [reply freemocap/fmc_origin_alignment.py:39]: yeah, right on. It does seem to do the job, so it's good enough for now :)

**Pull request merged**